src/main/java/com/researchspace/api/v1/controller/GalleryFilestoresApiController.java

```diff
@@ -0,0 +1,217 @@
+package com.researchspace.api.v1.controller;
+
+import com.researchspace.api.v1.GalleryFilestoresApi;
+import com.researchspace.model.DeploymentPropertyType;
+import com.researchspace.model.User;
+import com.researchspace.model.netfiles.NfsFileStore;
+import com.researchspace.model.netfiles.NfsFileStoreInfo;
+import com.researchspace.model.netfiles.NfsFileSystem;
+import com.researchspace.model.netfiles.NfsFileSystemInfo;
+import com.researchspace.netfiles.ApiNfsCredentials;
+import com.researchspace.netfiles.ApiNfsRemotePathBrowseResult;
+import com.researchspace.netfiles.NfsClient;
+import com.researchspace.netfiles.NfsFileDetails;
+import com.researchspace.netfiles.NfsFileTreeNode;
+import com.researchspace.netfiles.NfsTarget;
+import com.researchspace.service.NfsFileHandler;
+import com.researchspace.service.NfsManager;
+import com.researchspace.webapp.controller.DeploymentProperty;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.Valid;
+import javax.ws.rs.NotFoundException;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.validation.BindException;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestAttribute;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Slf4j
+@NoArgsConstructor
+@ApiController
+public class GalleryFilestoresApiController extends GalleryFilestoresBaseApiController
+    implements GalleryFilestoresApi {
+
+  @Autowired private NfsManager nfsManager;
+  @Autowired private NfsFileHandler nfsFileHandler;
+
+  @Override
+  public List<NfsFileStoreInfo> getUserFilestores(@RequestAttribute(name = "user") User user) {
```

Comment:
if you return a `Set` then you can avoid to scan the list to verify if the name is already existing